ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed

ccl/backupccl.TestProtectedTimestampsFailDueToLimits [failed](https://tanzanite.cluster.engflow.com/invocations/default/66620894-b422-4d30-985e-e4011ea6250b?testReportRun=1&testReportShard=39&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [8551145a0c99c4c95a28ec470e699d0c20ca97ab](https://github.com/cockroachdb/cockroach/commits/8551145a0c99c4c95a28ec470e699d0c20ca97ab):


```
=== RUN   TestProtectedTimestampsFailDueToLimits
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestProtectedTimestampsFailDueToLimits2206640277
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    backup_test.go:6453: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:6453
        	Error:      	Error message not equal:
        	            	expected: "pq: protectedts: limit exceeded: 0+30 > 1 bytes"
        	            	actual  : "pq: protectedts: limit exceeded: 0+31 > 1 bytes"
        	Test:       	TestProtectedTimestampsFailDueToLimits
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestProtectedTimestampsFailDueToLimits2206640277
--- FAIL: TestProtectedTimestampsFailDueToLimits (4.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=39</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #129976 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-24.2 release-blocker]
- #129975 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-24.2.1-rc release-blocker]
- #129972 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-23.1.26-rc release-blocker]
- #129971 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-23.1 release-blocker]
- #129970 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-23.2.10-rc release-blocker]
- #129969 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-23.2 release-blocker]
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestampsFailDueToLimits.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41820

ccl/backupccl.TestProtectedTimestampsFailDueToLimits [failed](https://tanzanite.cluster.engflow.com/invocations/default/b5443198-ca4d-41d5-9fc1-150c3933c616?testReportRun=1&testReportShard=39&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [8551145a0c99c4c95a28ec470e699d0c20ca97ab](https://github.com/cockroachdb/cockroach/commits/8551145a0c99c4c95a28ec470e699d0c20ca97ab):


```
=== RUN   TestProtectedTimestampsFailDueToLimits
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestProtectedTimestampsFailDueToLimits872323022
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    backup_test.go:6453: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_test.go:6453
        	Error:      	Error message not equal:
        	            	expected: "pq: protectedts: limit exceeded: 0+30 > 1 bytes"
        	            	actual  : "pq: protectedts: limit exceeded: 0+31 > 1 bytes"
        	Test:       	TestProtectedTimestampsFailDueToLimits
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestProtectedTimestampsFailDueToLimits872323022
--- FAIL: TestProtectedTimestampsFailDueToLimits (1.03s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=39</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #129979 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-24.1.4-rc]
- #129978 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-24.1]
- #129976 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-24.2]
- #129975 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-24.2.1-rc]
- #129972 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-23.1.26-rc]
- #129971 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-23.1]
- #129970 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-23.2.10-rc]
- #129969 ccl/backupccl: TestProtectedTimestampsFailDueToLimits failed [C-test-failure O-robot T-disaster-recovery branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProtectedTimestampsFailDueToLimits.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -6450,7 +6450,7 @@ func TestProtectedTimestampsFailDueToLimits(t *testing.T) {
 	// Creating the protected timestamp record should fail because there are too
 	// many spans. Ensure that we get the appropriate error.
 	_, err := db.Exec(`BACKUP TABLE foo, bar TO 'nodelocal://1/foo/byte-limit'`)
-	require.EqualError(t, err, "pq: protectedts: limit exceeded: 0+30 > 1 bytes")
+	require.ErrorContains(t, err, "pq: protectedts: limit exceeded")
 
 	// TODO(adityamaru): Remove in 22.2 once no records protect spans.
 	t.Run("deprecated-spans-limit", func(t *testing.T) {

